### DISCUSSION TITLE
Piccap hyperhdr webos lut for hdr ?

### DISCUSSION THREAD

**Discussion by todely** [awawa-dev/HyperHDR#384] [Q&A]
Hello
I have an lg oled rooted with piccap + hyperhdr webos and a raspberry with hyperhdr installed and driving led with d1 mini.
The system work great but:

1/ for HDR content, default LUT seems to not be the best one… how to replace it ?

2/ Does HDR content can be auto detected in piccap ou webos hyperhdr to applied LUT ?

Thanks !

**Comment by awawa-dev** [comment]
Hi
The webOS capturing has own limitation and the video quality is much lower than using typical USB grabber solution and that's based on results from the automatic LUT calibrator for HDR. At least for now. We can only adjust it using custom LUT tables and it improves it a bit but don't expect miracles: there is no science/formulas behind this but rather only manual adjustment. You can build your own LUT tables using internal (automatic LUT calibrator or LUT creator) or external tools (import from autodesk LUT file format). I don't know current status HyperHDR on webos but in the past there was HDR detection working in the webos backend and currently HyperHDR v19beta JSON API supports even multiple LUT (for HDR/SDR/DV what you like). You must ask about that somewhere else because webos is not officially supported here and I don't follow recent development since the root was patched by LG...and in a very clever way, even if it's possible to root the LG now has a control of early boot sequence which means e.g. updates can not be blocked any longer by 3th party scripts.

**Reply by todely** [reply]
Hello
i just using piccap, hyperhdr is on my raspberry, i didn’t need hyperhdr webos.
There is a method to get root on lg and stay root even with tv update.
It works pretty well between piccap and hyperhdr ! Hdr global command automatically go on when hdr content is displayed on tv and color accuracy is not so bad, no lag (tv and raspberry on ethernet)
Where i can fin hyperhdr v19beta ?

thanks for all your work once again !

**Reply by awawa-dev** [reply]
You can also try this LUT table made for webos but the result may depend on your TV model or the video source (may work for HDR but not for DV or vice versa) which shows we dont fully understand what's happening with the capturing on webos. Nevertheless it worked pretty well for HDR in most cases. If your capturing stream is very dark in the HyperHDR preview  that means it doesnt work in your case.     

https://github.com/awawa-dev/HyperHDR/raw/7bfbec468001c24ea97ec781797d769f6d1cd1f0/resources/lut/for_webos_hdr_flat_lut_lin_tables.tar.xz

**Comment by awawa-dev** [comment]
Yes, in most cases your root will stay after the update but probably SSH access will be lost. There is also latest crashd method but it doesn't change the fact that the boot sequence is locked now. If your TV is without a connection to the internet (or advanced firewall/pi-hole rules are applied to prevent it) then probably it doesn't matter. Otherwise an accidentally (or forced) update may break your TV or LG could acquire some statistics information (and void the warranty if applied). Everyone has to decide for themselves whether or not to root the system using current methods.  

I've seen also reports that suggest very high average system overload even if the load for the application processes like weboscapturing is low. If we assume that TV CPU is 4 cores unit, the average were between 2 to 3 per core (8-12 overall). But ```top``` report was also confusing because the fourth core never worked (blocked by LG webos or power saving feature?). So then the average load was between 3 to 4 per core and you can google it what it means. If you have HyperHDR installed externally on Rpi it should be lower, but also suggest you to verify these statistics.

Currently this is latest beta https://github.com/awawa-dev/HyperHDR/actions/runs/3293485210 (installers from the latest build are kept in the github action)

**Comment by todely** [comment]
with crashd method boot sequence is not looked, autostart works (and SSH too), i have update today firmware without loosing SSH / telnet / autostart.
I have check with top and cpu usage is between 100-105% (i have oled BX)
I will try to lower capture resolution to see if change
There is an open issue about this high CPU usage i will follow it.

![image](https://user-images.githubusercontent.com/11060244/197586176-718145ed-8712-4f21-ac63-d85289eb9823.png)

**Comment by awawa-dev** [comment]
Dont look at these 100-102% (statistics can be over 100% as it's multi-core unit press ```1```  to see them) because your TV is probably frying elsewhere...unless it's a bug. Christ, never seen something like that over 30:
![obraz](https://user-images.githubusercontent.com/69086569/197587224-e80ab14f-5aa2-4a3b-bb96-16e15b8c38de.png)
I would stop using it unless you figure it out what's wrong (frying CPU or a bug in top)

**Comment by todely** [comment]
After change few parametre (not capturing UI, lower resolution)
![image](https://user-images.githubusercontent.com/11060244/197587864-1e78a56d-32e1-44c1-8a58-3457c040f713.png)

**Reply by awawa-dev** [reply]
not much better, take a look at the load average: short(39)/mid(21) term are incredibly high and dangerous and long term is chasing them with time.

**Reply by todely** [reply]
After reboot:
![image](https://user-images.githubusercontent.com/11060244/197589052-cc5604fe-7e30-4846-8054-c19d8c4f6aca.png)

I will try to check without piccap running

**Reply by todely** [reply]
With piccap stopped the same:
![image](https://user-images.githubusercontent.com/11060244/197589510-6cd91c9e-051f-4a96-b542-88151a359987.png)

**Reply by awawa-dev** [reply]
Give it some time or/and disable the weboscapturing at the TV boot and restart the TV. Also press ```1``` to shows cores. If you have an access to ```htop``` then run it. I think I saw it in the webos discord some time ago. I really don't know what's going on with your system or it's some kind of ```top``` bug (but in these case you can't trust also in low load of webos capturing process).

**Reply by todely** [reply]
htop does not exist.
I have uncheck "autostart" and reboot 

![image](https://user-images.githubusercontent.com/11060244/197591289-39f6a4bf-26b5-4620-8c12-1a1bd48c908a.png)

Tv is displaying video coming from HDMI so perhaps it's normal ?

**Reply by awawa-dev** [reply]
I think you should ask for help on the openlgtv discord. Some reports from top were posted there but I think they never exceeded 12. Yours are over 30.  You should also find the ```htop``` package there.

**Reply by todely** [reply]
ok i will ask on discord about this

**Reply by todely** [reply]
i have full reset TV and only use crashd method, top command from telnet session give me same result... i think it's a firmware bug perhaps ? LG have deploy 2 update in 1 week..
Anyway to check load average without applied crashd ?

**Reply by awawa-dev** [reply]
I think LG has a legit way to obtain root for developer purposes, but it's transient and work only for a day or two and then you must repeat the procedure. Probably must be disabled before you apply root using other way. There can be multiply reasons for such high average load, most likely I/O bottleneck or some light processes running wild and hammering the system.

**Reply by todely** [reply]
I have install offical tools from LG to monitor system.. so TV full reset, there is ONLY LG DEV APP installed:
![image](https://user-images.githubusercontent.com/11060244/198014408-4d49f699-3165-4195-a976-50edd98812d7.png)

![image](https://user-images.githubusercontent.com/11060244/198014589-df23d0be-061d-4dcc-a156-bdba7654942b.png)

![image](https://user-images.githubusercontent.com/11060244/198015446-fe98b310-8e82-4f8a-b5e6-bb7e98e6a967.png)

![image](https://user-images.githubusercontent.com/11060244/198015475-bb8c8790-1b00-461f-9929-ce07bf69b39c.png)
